docs/content/developers/design-keuzes.md

```diff
@@ -190,3 +190,61 @@ RGBZ. BESLUIT krijgt een optionele relatie naar ZAAK.
 
 De verwachting is dat er later vergelijkbare designkeuzes gemaakt worden voor
 andere resources, zoals bijvoorbeeld klantcontacten.
+
+
+## Response bij input validatie
+
+De DSO schrijft een structuur voor waar een fout-response aan moet voldoen
+in het geval van validatiefouten:
+
+```json
+{
+    "type": "...",
+    "title": "...",
+    "status": 400,
+    "detail": "...",
+    "instance": "...",
+    "invalid-params": [{
+        "type": "...",
+        "name": "<veldnaam>",
+        "reason": "Maximale lengte overschreden.",
+    }]
+}
+```
+
+Op het [forum](https://forum.pdok.nl/t/formaat-foutafhandeling-input-validatie-api-50/1848)
+is nagevraagd hoe er moet omgegaan worden met meerdere fouten op hetzelfde veld,
+met de conclusie dat dit beter gespecifieerd moet worden.
+
+In ZDS 2.0 kiezen we ervoor om elke fout op een veld als apart object op te
+nemen binnen de `"invalid-params"` sleutel. Dit laat clients toe om elke
+individuele fout te renderen zoals zij wensen.
+
+Tevens voegen we op hoofdniveau van fouten en binnen een object in
+`"invalid-params"` een sleutel `"code"` toe. Dit veld bevat een waarde die door
```

Comment:
Wellicht "code" dan ook opnemen in voorbeeld onderaan.

Reply:
Dat was de bedoeling, maar ben ik zo te zien vergeten. Pas ik maandag aan!

Reply:
Zijn er standaard codes of gaan we een lijst met eigen codes maken? In StUF waren er eigen foutcodes, eventueel zouden we die, waar zinvol, kunnen overnemen?

Reply:
@michielverhoef hrm, goeie vraag. Indien er een mogelijke set van codes komt, zou die idealiter ook in de API spec opgenomen moeten worden. Echter, dat zou ik nog wat naar achteren schuiven tot we effectief wat validatie hebben.

De natuur van de codes is wel een stuk technischer _en_ specifieker dan in StUF. Ik denk dat we eerst maar even moeten kijken hoe ver we komen, dit vastleggen in integratietests en dan een volgende sprint een inventaris maken van de codes en of er bepaalde types/klasses van fouten zijn.

Reply:
Hierboven staat een "template" van een error volgens de DSO. Het veld code is toegevoegd door ons, en dat is het voorbeeld (concreet met data) beneden.

Je hebt de PR goedgekeurd dus ik ga er van uit dat het niet blokkerend is.

Reply:
Daaraan voeg ik nog toe dat `type` optioneel is volgens de RFC die DSO gebruikt :)

Reply:
Het kwartje was inmiddels gevallen en ik had mijn eigen vraag eerder al beantwoord via Slack aan Sergei. Dus inderdaad, niet blokkerend.